integration-tests/features/steps/stack_analyses_v2.py

```diff
@@ -0,0 +1,378 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Author: Dharmendra G Patel <dhpatel@redhat.com>
+#
+"""Tests for API endpoints that performs stack analysis V2 API."""
+import os
+import time
+import logging
+import requests
+from behave import when, then
+from urllib.parse import urljoin
+from src.attribute_checks import check_attribute_presence
+from src.parsing import parse_token_clause, parse_valid_clause
+from src.attribute_checks import check_timestamp
+from src.json_utils import get_value_using_path
+from src.stack_analysis_common import get_json_data, check_frequency_count
+
+
+logger = logging.getLogger(__file__)
+
+ECOSYSTEM_TO_MANIFEST_NAME_MAP = {
+    'pypi': 'pylist.json',
+    'npm': 'npmlist.json',
+    'maven': 'dependencies.txt'
+}
+
+
+def sav2_get_endpoint(context):
+    """Return endpoint for the stack analysis v2."""
+    return urljoin(context.threescale_preview_url, '/api/v2/stack-analyses/')
+
+
+def sav2_post_request(context, ecosystem, manifest, with_user_key, is_valid):
+    """Send stack analyses v2 post request based on params."""
+    logger.debug('ecosystem: {} manifest: {} with_user_key: {} '
+                 'is_valid: {}'.format(ecosystem, manifest, with_user_key, is_valid))
+    context.manifest = manifest
+
+    # set default values
+    files = {}
+    data = {}
+
+    # Add ecosystem if not None
+    if ecosystem != 'None':
+        data['ecosystem'] = ecosystem
+
+    # Read manifest if not None
+    if manifest != 'None':
+        filename = 'data/{}'.format(manifest)
+        files['manifest'] = (ECOSYSTEM_TO_MANIFEST_NAME_MAP.get(ecosystem, 'invalid_name.json'),
+                             open(filename, 'rb'))
+        data['file_path'] = os.path.abspath(os.path.dirname(filename))
+
+    if with_user_key:
+        if is_valid:
+            params = {'user_key': context.three_scale_preview_user_key}
+        else:
+            params = {'user_key': 'INVALID_USER_KEY_FOR_TESTING'}
+        logger.debug('POST {} files: {} data: {} params: {}'.format(sav2_get_endpoint(context),
+                                                                    files, data, params))
+        response = requests.post(sav2_get_endpoint(context), files=files, data=data, params=params)
+    else:
+        response = requests.post(sav2_get_endpoint(context), files=files, data=data)
+
+    logger.debug('status_code: {} response: {}'.format(response.status_code, response.text))
+    context.response = response
+
+
+def sav2_get_analyzed_components(context):
+    """Return all analyzed components from the deserialized JSON file."""
+    json_data = get_json_data(context)
+
+    components = get_value_using_path(json_data, 'analyzed_dependencies')
+    assert components is not None
+
+    return components
+
+
+def sav2_get_components_with_cve(components):
+    """Get all components with CVE record(s)."""
+    result = []
+
+    for component in components:
```

Comment:
is there any other way to these we are just increasing our cyclomatic complexity here ?